File: app/src/main/java/org/getlantern/lantern/model/Bandwidth.java
package org.getlantern.lantern.model;

public class Bandwidth {
    private Long percent;
    private Long remaining;
    private Long allowed;

    public Bandwidth(Long percent, Long remaining, Long allowed) {
        this.percent = percent;
        this.remaining = remaining;
        this.allowed = allowed;
    }

    public Long getPercent() {
        return percent;
    }

    public Long getUsed() {
        return allowed - remaining;
    }

    public Long getRemaining() {
        return remaining;
    }

    public Long getAllowed() {
        return allowed;
    }

    public String toString() {
        return String.format("Bandwidth update: %d/%d (%d)",
                remaining, allowed, percent);
    }
}

File: app/src/main/java/org/getlantern/lantern/model/Bandwidth.kt
package org.getlantern.lantern.model

class Bandwidth(
        val percent: Long, // [0, 100]
        val remaining: Long, // in MB
        val allowed: Long, // in MB
        val ttlSeconds: Long // number of seconds left before data reset
) {

    val used: Long
        get() = allowed - remaining

    override fun toString(): String {
        return String.format("Bandwidth update: %d/%d (%d). TTL: %d (seconds)",
                remaining, allowed, percent, ttlSeconds)
    }
}